fingerprint/src/main/java/com/futronictech/Scanner.kt

```diff
@@ -0,0 +1,158 @@
+@file:Suppress("unused")
+package com.futronictech
+
+import com.fingerprint.scanner.FutronictechFingerprintScanner
+import java.io.File
+
+
+@Suppress("PrivatePropertyName")
+internal class Scanner(cacheDir: File) {
+    private var m_hDevice: Long = 0
+    private var m_ErrorCode = FTR_ERROR_NO_ERROR
+    private var m_NFIQ = 0
+    private var m_ImageHeight = 0
+    private var m_ImageWidth = 0
+    var isSyncDirInitialized = false
+
+    init {
+        isSyncDirInitialized = setGlobalSyncDir(cacheDir.absolutePath)
+    }
+
+    @JvmName("OpenDeviceCtx")
+    external fun openDeviceOnInterfaceUsbHost(obj: FutronictechFingerprintScanner): Boolean
+
+    @JvmName("CloseDevice")
+    external fun closeDevice(): Boolean
+
+    @JvmName("GetDiodesStatus")
+    external fun getDiodesStatus(bytes: ByteArray?): Boolean
+
+    @JvmName("GetFrame")
+    external fun getFrame(bytes: ByteArray?): Boolean
+
+    @JvmName("GetImage")
+    external fun getImage(i: Int, bytes: ByteArray?): Boolean
+
+    @JvmName("GetImage2")
+    external fun getImage2(i: Int, bytes: ByteArray?): Boolean
+
+    @JvmName("GetImageByVariableDose")
+    external fun getImageByVariableDose(i: Int, bytes: ByteArray?): Boolean
+
+    @JvmName("GetImageSize")
+    external fun getImageSize(): Boolean
+
+    @JvmName("GetInterfaces")
+    external fun getInterfaces(bytes: ByteArray?): Boolean
+
+    @JvmName("GetNfiqFromImage")
+    external fun getNfiqFromImage(bytes: ByteArray?, i: Int, i2: Int): Boolean
+
+    @JvmName("GetSerialNumber")
+    external fun getSerialNumber(): String?
+
+    @JvmName("GetVersionInfo")
+    external fun getVersionInfo(): String?
+
+    @JvmName("IsFingerPresent")
+    external fun isFingerPresent(): Boolean
+
+    @JvmName("OpenDevice")
+    external fun openDevice(): Boolean
+
+    @JvmName("OpenDeviceOnInterface")
+    external fun openDeviceOnInterface(i: Int): Boolean
+
+    @JvmName("Restore7Bytes")
+    external fun restore7Bytes(bytes: ByteArray?): Boolean
+
+    @JvmName("RestoreSecret7Bytes")
+    external fun restoreSecret7Bytes(inBytes: ByteArray?, outBytes: ByteArray?): Boolean
+
+    @JvmName("Save7Bytes")
+    external fun save7Bytes(bytes: ByteArray?): Boolean
+
+    @JvmName("SaveSecret7Bytes")
+    external fun saveSecret7Bytes(inBytes: ByteArray?, outBytes: ByteArray?): Boolean
+
+    @JvmName("SetDiodesStatus")
+    external fun setDiodesStatus(i: Int, i2: Int): Boolean
+
+    @JvmName("SetGlobalSyncDir")
+    external fun setGlobalSyncDir(str: String?): Boolean
+
+    @JvmName("SetLogOptions")
+    external fun setLogOptions(i: Int, i2: Int, str: String?): Boolean
+
+    @JvmName("SetNewAuthorizationCode")
+    external fun setNewAuthorizationCode(bytes: ByteArray?): Boolean
+
+    @JvmName("SetOptions")
+    external fun setOptions(i: Int, i2: Int): Boolean
+
+    val imageWidth: Int
```

Comment:
It appears that this is not used, same with a lot of other functions and variables, are they going to be used in the future?

Reply:
These functions are used in the c++ api